### PR TITLE
Clip invalid positions while seeking

### PR DESCRIPTION
**Pull request by as-cii**
As said on #3, code-wise this still needs some adjustments. However I wanted to demonstrate how we could sanitize `position` while seeking on `TransformIterator` and clip accordingly whenever needed.

Hopefully, this is the way to go: otherwise, well, let's just close this :laughing: 

/cc: @nathansobo @maxbrunsfeld

### CODE REVIEW

File: src/transform-layer.coffee [hunk 1]
```diff
@@ -64,52 +69,49 @@ class TransformLayerIterator
...
+  seekToSourcePosition: (position, clip) ->
+    calculateOvershoot = (position, lastPosition, lastSourcePosition) =>
+      overshoot = position.traversalFrom(lastSourcePosition)
+      lastPosition = lastPosition.traverse(overshoot)
+      @position = lastPosition
+      @sourcePosition = position
+
+    @_seek (=> @sourcePosition), position, calculateOvershoot, clip
+
+  _seek: (seekPropertyFn, position, calculateOvershoot, clip = CLIP_BACKWARD) ->
```
Comment (as-cii) [inline src/transform-layer.coffee]: I am not super happy with the naming adopted here: I feel like naming this method `seek` and renaming our current `seek` to `seekToTargetPosition` could be a nice solution.

What do you think?

**Comment by maxbrunsfeld** [comment]
Nice, thanks @as-cii!

File: src/transform-layer.coffee (hunk 1 above)
Comment (maxbrunsfeld) [inline src/transform-layer.coffee]: We think we should just keep the duplication for now. Another approach might be to extract a method for performing the clipping and/or the overshoot calculation. For now though, while things are still taking shape, I think the duplication is ok.

**Comment by as-cii** [comment]
@maxbrunsfeld: :+1: about the duplication comment.

By the way, this PR was against ns-clipping and therefore I think it didn't make it into master.

I am away from keyboard and I cannot fix it right now unfortunately: sorry for the misunderstanding 🙏

**Pull request merged**